File: sol/2401/2401.cpp
struct Solution {
    int longestNiceSubarray(vector<int>& nums) {
        int res = 1, n = nums.size(), mask = nums[0], l = 0;
        for (int r = 1; r < n; ++r){
            while (l < r && (mask & nums[r]) != 0){
                mask ^= nums[l++];
            }
            mask |= nums[r];
            res = max(res, r - l + 1);
        }
        return res;
    }
};

File: sol/2401/2401.py
class Solution:
    def longestNiceSubarray(self, nums: List[int]) -> int:
        res = 1
        n = len(nums)
        mask = nums[0]
        l = 0
        for r in range(1, n):
            while l < r and (mask & nums[r]) != 0:
                mask ^= nums[l]
                l += 1
            mask |= nums[r]
            res = max(res, r - l + 1)
        return res
